store/src/java/com/zimbra/cs/imap/ImapPath.java

```diff
@@ -284,15 +283,39 @@ protected ImapMailboxStore getOwnerImapMailboxStore(boolean forceRemote) throws
             } else if (mCredentials == null) {
                 imapMboxStore = null;
             } else {
-                ZMailbox zmbox = getOwnerZMailbox();
-                imapMboxStore = (null == zmbox) ? null : ImapMailboxStore.get(zmbox, this.getOwnerAccountId());
+                imapMboxStore = getRemoteImapMailboxStoreForOwner();
             }
         }
         return imapMboxStore;
     }
 
-    private String accountIdFromCredentials() {
-        return (mCredentials == null) ? null : mCredentials.getAccountId();
+    private ImapMailboxStore getRemoteImapMailboxStoreForOwner() throws ServiceException {
+        ZMailbox zmbox = getOwnerZMailbox();
+        return (null == zmbox) ? null : ImapMailboxStore.get(zmbox, this.getOwnerAccountId());
+    }
+
+    private ZMailbox getZMailboxForAccount(Account target) throws ServiceException {
+        Account acct = mCredentials == null ? null :
+            Provisioning.getInstance().get(AccountBy.id, mCredentials.getAccountId());
```

Comment:
I'd been wondering if we allowed ternary operators. Glad to see it's ok since they are handy.